src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets

```diff
@@ -211,7 +217,9 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- TODO perform any preprocess transforms on the files -->
 
     <ItemGroup>
-      <ResolvedAssembliesToPublish Include="@(ReferenceCopyLocalPaths)" Exclude="@(ResolvedAssembliesToPublish)">
+      <ResolvedAssembliesToPublish Include="@(ReferenceCopyLocalPaths)"
+                                   Exclude="@(ResolvedAssembliesToPublish)"
+                                   Condition="'$(PublishProjectReferenceDocumentationFiles)' == 'true' or '%(Extension)' != '.xml'">
```

Comment:
Here's how the documentation file works:

1. It starts life as a property `<DocumentationFile />` defined in an individual project
2. It gets added to an item group `@(DocFileItem)` [here](https://github.com/Microsoft/msbuild/blob/5463237b0c014e9f3e0f1c5157780a8f40545ed3/src/Tasks/Microsoft.CSharp.CurrentVersion.targets#L156)
3. `@(DocFileItem)` is passed to `<Csc />` [here](https://github.com/Microsoft/msbuild/blob/5463237b0c014e9f3e0f1c5157780a8f40545ed3/src/Tasks/Microsoft.CSharp.CurrentVersion.targets#L261)
4. It is added an item group `@(FinalDocFile)` [here](https://github.com/Microsoft/msbuild/blob/2634f05a660b14c64ae4e924d900dd200d5032e7/src/Tasks/Microsoft.Common.CurrentVersion.targets#L350) which is a full path to where the file should be copied to
5. The `@(DocFileItem)` is copied to `@(FinalDocFile)` [here](https://github.com/Microsoft/msbuild/blob/2634f05a660b14c64ae4e924d900dd200d5032e7/src/Tasks/Microsoft.Common.CurrentVersion.targets#L4062)
6. `<ResolveAssemblyReference />` gets passed a [list of related file extensions](https://github.com/Microsoft/msbuild/blob/2634f05a660b14c64ae4e924d900dd200d5032e7/src/Tasks/Microsoft.Common.CurrentVersion.targets#L578) which includes `.xml`.

So technically, the only reason the Doc file shows up in `@(ReferenceCopyLocalPaths)` is because it has the same file name without extension as the output assembly.  [This logic](https://github.com/Microsoft/msbuild/blob/6851538897f5d7b08024a6d8435bc44be5869e53/src/Tasks/AssemblyDependency/ReferenceTable.cs#L879) takes the output file name and looks for "related" files by extensions, one of which is `.xml`.  If people were trying to be fancy, they could have their documentation file named something else and it would show up in `@(ReferenceCopyLocalPaths)`.

That said, it's probably safe to assume that the only related file ending in `.xml` is the doc file.  MSBuild does not currently translate any metadata to these related files that indicate what they were originally.